app/models/event.rb

```diff
@@ -48,14 +48,14 @@ class Event < ApplicationRecord
 	belongs_to :nonprofit
 	belongs_to :profile
 	has_many :donations
-	has_many :charges, through: :tickets
-	has_many :ticketholders, through: :tickets, source: :supporter
 	has_many :supporters, through: :donations
 	has_many :recurring_donations
 	has_many :ticket_levels, :dependent => :destroy
   has_many :event_discounts, dependent: :destroy
 	has_many :tickets
 	has_many :payments, through: :tickets
+	has_many :charges, through: :tickets
```

Comment:
Rails 5.1 doesn't like if you reference an association in a "through" before the association is defined. This fixes it.

Reply:
In general, I'd say because it's not related to this PR. I'm happy to do it in a separate one.